### DISCUSSION TITLE
Ensemble yolov5

### DISCUSSION THREAD

**Discussion by CesareDavidePace** [obss/sahi#740] [Q&A] [answered]
Hi,
There is a possibility to use ensemble with yolov5?

**Comment by fcakyon** [comment]
this should work:

```python
from sahi import AutoDetectionModel

detection_model = AutoDetectionModel.from_pretrained(
    model_type="yolov5",
    model_path = ["weight1.pt", "weight2.pt"],
    confidence threshold = 0.3
)
```

then perform prediction using this model.

**Reply by CesareDavidePace** [reply]
i have this error:

('model_path is not a valid yolov5 model path: ', Exception("Not a yolov5 model: <class 'yolov5.models.experimental.Ensemble'>"))

**Reply by fcakyon** [reply]
Then, unfortunately, we dont support it. You can write your custom `Yolov5EnsembleDetectionModel` implementing `DetectionModel` class to add such support.

You can follow the steps given in readme: https://github.com/obss/sahi#contributing

**Comment by fcakyon** [comment, marked as answer]
Currently not supported but we are open to PRs :+1:
